cameraview/src/main/java/com/otaliastudios/cameraview/CameraOptions.java

```diff
@@ -469,4 +485,15 @@ public float getExposureCorrectionMinValue() {
     public float getExposureCorrectionMaxValue() {
         return exposureCorrectionMaxValue;
     }
+
+    /**
+     * The maximum value of digital zoom available by Hardware.
+     * This is presumably 1 and above than 1 in case zoom is avaible.
+     *
+     * @return max zoom value
+     */
+    @SuppressWarnings("WeakerAccess")
+    public float getMaxZoomValue() {
```

Comment:
In CameraView, zoom goes from 0 to 1 always. So setZoom(0) means remove any zoom, and setZoom(1) means go to this maxZoomValue.

So in the engine you should map the 0-1 external range to the Camera2 internal range. And then you can remove this method!

Reply:
I mapped the values in Camera2Engine according to the range 0.0-1.0 using 

zoom * mCameraOptions.getMaxZoomValue()

but we need the max zoom possible value to create the Rect we are cropping on every zoom change.

I can alternatively remove this method and can query on CameraCharacteristics every time but I think that will be tedious work. Instead, querying it once and saving it to a variable will fast our zooming and Rect creation

Reply:
I'm not against storing the variable, I just don't want to expose this to public since it has no public meaning... You could move this to Camera2Engine class.

About the zoom, I think you want to map:
externalZoom = 0 -> internalZoom = 1
externalZoom = 1 -> internalZoom = maxZoom

So I think the formula should be `externalZoom = 1 + internalZoom * (maxZoom - 1)` instead of `externalZoom = internalZoom * maxZoom`